glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -97,12 +97,47 @@ open class GleanInternalAPI internal constructor () {
         }
     }
 
+    /**
+     * Enable or disable Glean collection and upload.
+     *
+     * Metric collection is enabled by default.
+     *
+     * When uploading is disabled, metrics aren't recorded at all and no data
+     * is uploaded.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setUploadEnabled(enabled: Boolean) {
+        // logger.info("Metrics enabled: $enabled")
+        // val origUploadEnabled = uploadEnabled
+        // uploadEnabled = enabled
+        // if (isInitialized() && origUploadEnabled != enabled) {
+        //     onChangeUploadEnabled(enabled)
+        // }
+        // TODO: stub
+    }
+
+    /**
+     * Get whether or not Glean is allowed to record and upload data.
+     */
+    fun getUploadEnabled(): Boolean {
```

Comment:
We have `glean_is_upload_enabled`....

Now that I commented on the second thing here I think it's best to fill them in a second step and not here.

Reply:
I agree :) Additionally, let's add a log line :D